src/sentry/integrations/slack/sdk_client.py

```diff
@@ -0,0 +1,24 @@
+from slack_sdk import WebClient
+
+from sentry.models.integrations import Integration
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.silo.base import SiloMode
+
+
+class SlackClient(WebClient):
+    def __init__(self, integration_id: int, base_url: str | None = None):
+        integration = None
+        if SiloMode.get_current_mode() == SiloMode.REGION:
+            integration = integration_service.get_integration(integration_id=integration_id)
+        else:  # control or monolith (local)
+            integration = Integration.objects.filter(id=integration_id).first()
+
+        if integration is None:
+            raise ValueError(f"Integration with id {integration_id} not found")
```

Comment:
we don't raise any errors if initialization fails in the existing `SlackClient`. all subsequent calls are no-ops, but i think we should fail loudly so we're aware something is broken

Reply:
+1